fvm/src/kernel/mod.rs

```diff
@@ -226,6 +221,12 @@ pub trait GasOps {
     /// `name` provides information about gas charging point
     fn charge_gas(&mut self, name: &str, compute: i64) -> Result<()>;
 
+    /// Returns available gas.
+    fn borrow_milligas(&mut self) -> Result<i64>;
+
+    /// Sets available gas to a new value, creating a gas charge if needed
+    fn return_milligas(&mut self, name: &str, newgas: i64) -> Result<()>;
+
```

Comment:
These traits usually map 1:1 with syscall namespaces. Can you move these up to the Kernel trait itself? Actually: do they even _need_ to be part of the trait? Can't they just be implemented in DefaultKernel?

Reply:
- They need to be a part of the trait because we call them in the CallManager and InvocationContext.
- IMO, splitting gas logic into two places would be more confusing (although I wish we didn't need these functions here).